packages/react/src/ReactElement.js

```diff
@@ -21,28 +21,24 @@ const RESERVED_PROPS = {
 
 let specialPropKeyWarningShown, specialPropRefWarningShown;
 
-function hasValidRef(config) {
+function hasValidProp(config, prop) {
   if (__DEV__) {
-    if (hasOwnProperty.call(config, 'ref')) {
-      const getter = Object.getOwnPropertyDescriptor(config, 'ref').get;
+    if (hasOwnProperty.call(config, prop)) {
+      const getter = Object.getOwnPropertyDescriptor(config, prop).get;
       if (getter && getter.isReactWarning) {
         return false;
       }
     }
   }
-  return config.ref !== undefined;
```

Comment:
Thanks, but it is important that this stays as a direct access so JS engines can better optimize what this code is doing in production.

Reply:
Ohh, got it! Thanks for feedback. And what about object arguments and destructuring should I merge them?

Reply:
I don't think these are important there. We generally prefer less terse code so we avoid surprises from Babel when it generates suboptimal code.